openflow13/match.go

```diff
@@ -1323,3 +1330,100 @@ func NewTunnelIpv4DstField(tunnelIpDst net.IP, tunnelIpDstMask *net.IP) *MatchFi
 
 	return f
 }
+
+// ARP Host Address field message, used by arp_sha and arp_tha match
+type ArpXHaField struct {
+	arpHa net.HardwareAddr
+}
+
+func (m *ArpXHaField) Len() uint16 {
+	return 6
+}
+func (m *ArpXHaField) MarshalBinary() (data []byte, err error) {
+	data = make([]byte, m.Len())
+	copy(data, m.arpHa)
+	return
+}
+
+func (m *ArpXHaField) UnmarshalBinary(data []byte) error {
+	if len(data) < int(m.Len()) {
+		return errors.New("The byte array has wrong size to unmarshal ArpXHaField message")
+	}
+	copy(m.arpHa, data[:6])
+	return nil
+}
+
+func NewArpThaField(arpTha net.HardwareAddr) *MatchField {
```

Comment:
I think that this function and NewArpShaField() are the same code except for the OXM_FIELD_ARP_* constant. I would suggest refactoring them to use a common helper function.